src/main/scala/fpinscalalib/FunctionalDataStructuresSection.scala

```diff
@@ -0,0 +1,634 @@
+package fpinscalalib
+
+import org.scalatest.{FlatSpec, Matchers}
+import fpinscalalib.List._
+import fpinscalalib.Tree._
+
+/** @param name functional_data_structures
+  */
+object FunctionalDataStructuresSection extends FlatSpec with Matchers with org.scalaexercises.definitions.Section {
+
+  /**
+    * = Singly linked lists =
+    *
+    * First let's examine what's probably the most ubiquitous functional data structure, the singly linked list. The
```

Comment:
Long lines, IMHO, we'd consider split them in shorter ones to facilitates the reading. What do you think?